apps/web/src/pages/templates/editor/TourProvider.tsx

```diff
@@ -0,0 +1,89 @@
+import { useMantineColorScheme } from '@mantine/core';
+import React, { useMemo, useCallback } from 'react';
+import Joyride from 'react-joyride';
+import { colors } from '../../../design-system';
+
+import { useDigestWorkflowTour } from './useDigestWorkflowTour';
+
+const TourContext = React.createContext({
+  startTour: () => {},
+  stopTour: () => {},
+  stepTourIndex: 0,
+  setStep: (index) => {},
+});
+
+export const useTour = () => React.useContext(TourContext);
+
+export const TourProvider = React.memo(() => {
+  const [{ run, stepTourIndex }, setState] = React.useState<{ run: boolean; stepTourIndex: number }>({
+    run: false,
+    stepTourIndex: 0,
+  });
+  const { colorScheme } = useMantineColorScheme();
+
+  const startTour = useCallback(() => {
+    setTimeout(() => setState((old) => ({ ...old, run: true })), 0);
+  }, []);
+
+  const stopTour = useCallback(() => {
+    setTimeout(() => setState((old) => ({ ...old, run: false })), 0);
+  }, []);
+
+  const value = useMemo(
+    () => ({
+      startTour,
+      stopTour,
+      stepTourIndex,
+      setStep: (index) => {
+        // give the time for the target to be rendered
+        setState((old) => ({ ...old, stepTourIndex: index, run: true }));
+      },
+    }),
+    [startTour, stopTour, stepTourIndex]
+  );
+
+  const { digestTourSteps } = useDigestWorkflowTour({ startTour });
```

Comment:
This hook returns the steps that are specific to the digest tour. 
In the future, we might have other hooks that will return other hint steps, for ex. for the editing email. 
And we won't need to implement the logic from scratch, just we would have to pass those steps to the `Joyride` component.

Reply:
Do you mean that it is possible to concat here several tours? one after another?

Reply:
what I meant is that you can write any other hook that will return different tour steps, and then pass these steps to the tour provider... it will a little logic here